dev/AppNotifications/AppNotifications.idl

```diff
@@ -6,7 +6,6 @@ import "..\AppLifecycle\AppLifecycle.idl";
 namespace Microsoft.Windows.AppNotifications
 {
     // Event args for the Notification Activation
-    [experimental]
     runtimeclass AppNotificationActivatedEventArgs
```

Comment:
Can this PR also be used to apply the appropriate contract information?  @DrusTheAxe or @dhoehna might have guidance for you on the specific one to use.

Reply:
I agree with @jonwis.  This PR can be used to add contracts.  

The questions is this "Should a contract be added to this runtime class, or something higher up?"

Reply:
Happy to do that. @dhoehna - What do we need to decide if we apply the contract to this class or to a higher  up?

What do contracts look like? Can you point me to an example?

Reply:
An example is here: https://github.com/microsoft/WindowsAppSDK/pull/2110#pullrequestreview-885233578

@jonwis @DrusTheAxe what level should this new contract be placed?

Reply:
Sorry, what do you mean by "level"?

What is "higher up" than a (runtime)class? A namespace? I didn't think you could apply a contract to a namespace, and even if you _could_ I'm not sure you _should_. Would that mean everything in namespace A.B gets it...unless specifically overridden per type? Would child namespaces inherit it too? MIDL doesn't tend to like varying definition of a type across...an .idl? Compilation? e.g.

```
[contract(FooContract, 1)]
namespace Foo
{
    [contractversion(1)]
    apicontract FooContract;

    [contract(FooContract, 1)]
    runtimeclass Bar {...}
}
```

MIDL historically doesn't like undefined type references that are later defined, as in this example **line 1**. I'm doubtful if this compiles.

Namespaces aren't types as I understand it. If so then I wouldn't expect contracts to apply to namespaces.

runtimeclass, interface, enum, more are types and be have a contract applied.

Do the MIDL docs clearly define what `[contract(...)]` can be applied to? If not that would be a doc bug.

Do the MIDL(?) docs provide _guidance_ onw how you could _and should_ use them? As we've seen contracts are a useful versioning mechanism for more than OS types. @BenJKuhn guidance on when/how to use them may be thin but if so, we should improve that. Need a doc bug?

Reply:
> What is "higher up" than a (runtime)class? 

I don't know actually.  I thought that there was something higher up than a runtime class.  My mistake.

@loneursid Please add a contract to AppNotifications and call it WindowsAppNotificationsContract.  Start with version 1.

Reply:
@dhoehna - I'm getting this error: **error MIDL2025: [msg]syntax error [context]: expecting { near "["**

Here's what I've tired: (based on your example)
```
namespace Microsoft.Windows.AppNotifications
{
    [contractversion(1)]
    apicontract WindowsAppNotificationsContract

    // Event args for the Notification Activation
    [Contract(MicrosoftWindowsAppNotifications, 1)]
    runtimeclass AppNotificationActivatedEventArgs

...
```

Reply:
Should have braces, not a semi-colon.

```
[contractversion(1)]
apicontract WindowsAppNotificationsContract { }
```

Reply:
Sorry about that.  The example has been updated.

Reply:
@dhoehna  - I have tagged the 3 runtime classes in this namespace with the contract. I haven't tagged the enums? Should I?

@DrusTheAxe - I've kept the initial name for the contract: WindowsAppNotificationsContract, let me know if I should change it.

Reply:
> WindowsAppNotificationsContract, let me know if I should change it.

IMO yes, as the 'Windows' part refers to the inbox Windows product types and not the appnotifications functionality. thus `namespace Microsoft.AppNotifications { apicontract AppNotificationsContract {} }`.

But it's an API Review question. You should ask your API Design representative.

Reply:
> I haven't tagged the enums

Yes